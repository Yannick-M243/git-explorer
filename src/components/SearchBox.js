import React, { Component } from 'react';
import GithubUsersList from './GithubUsersList.js';
import GitlabUsersList from './GitlabUsersList.js';
import FadeLoader from "react-spinners/FadeLoader";

class SearchBox extends Component {

    constructor(props) {
        super(props)
        this.handleInputChange = this.handleInputChange.bind(this);
        this.state = {
            input: "",
            githubUserResult: [],
            gitlabUserResult: [],
            githubLoaded: null,
            gitlabLoaded: null,
            error: null
        }
    }

    handleInputChange = (e) => {
        this.setState({ input: e.target.value });
    }

    searchUser = (e) => {
        e.preventDefault();

        if (this.state.input !== "") {
            this.setState({ githubLoaded: "loading" });
            this.setState({ gitlabLoaded: "loading" });
            this.searchUserOnGithub(this.state.input);
            this.searchUserOnGitLab(this.state.input);
        } else {
            alert("Please enter a username first!")
        }

    }

    searchUserOnGithub = (username) => {
        fetch(`/api/github/userinfo/${username}`)
            .then(res => res.json())
            .then(
                (result) => {
                    this.setState({
                        githubUserResult: result,
                        githubLoaded: "loaded"
                    });
                    console.log(result)
                },
                (error) => {
                    this.setState({
                        githubLoaded: "loaded",
                        error
                    });
                })
    }

    searchUserOnGitLab = (username) => {
        fetch(`/api/gitlab/userinfo/${username}`)
            .then(res => res.json())
            .then(
                (result) => {
                    this.setState({
                        gitlabUserResult: result,
                        gitlabLoaded: "loaded"
                    });
                    console.log(result)
                },
                (error) => {
                    this.setState({
                        gitlabLoaded: "loaded",
                        error
                    });
                })
    }

    render() {
        const { githubLoaded, gitlabLoaded, gitlabUserResult, githubUserResult, error } = this.state;
        return (
            <div className='container'>
                <div className='search-container' >
                    <form>
                        <input type="text" onChange={this.handleInputChange} />
                        <button type="button" onClick={this.searchUser} >Search</button>
                    </form>
                </div>
                {error ? <div className='user-container'><p>Error: {error.message}</p></div> : ''}
                <h3>Github:</h3>
                {githubLoaded === "loading" ? <FadeLoader color="black" /> : <GithubUsersList user={githubUserResult} githubLoaded={githubLoaded} />}
                <h3>Gitlab:</h3>
                {gitlabLoaded === "loading" ? <FadeLoader color="black" /> : <GitlabUsersList users={gitlabUserResult} gitlabLoaded={gitlabLoaded} />}
            </div>
        )
    }
}

export default SearchBox;