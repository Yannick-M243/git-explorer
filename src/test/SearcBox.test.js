import React from 'react';
import SearchBox from '../components/SearchBox.js';
import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer
        .create(<SearchBox/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});