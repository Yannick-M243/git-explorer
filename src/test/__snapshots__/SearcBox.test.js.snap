// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <h1>
    Git Explorer
  </h1>
  <div
    className="search-container"
  >
    <form>
      <input
        onChange={[Function]}
        onKeyDown={[Function]}
        type="text"
      />
      <button
        onClick={[Function]}
        placeholder="Enter a username"
        type="button"
      >
        Search
      </button>
    </form>
  </div>
  <h3>
     Github:
  </h3>
  <h3>
     Gitlab:
  </h3>
  <h3>
     Repositories:
  </h3>
  <h3>
     Commits:
  </h3>
</div>
`;
